src/vehicle/internals/update_factory.go

```diff
@@ -4,43 +4,101 @@ import (
 	"fmt"
 	"time"
 
+	"github.com/HyperloopUPV-H8/Backend-H8/common"
 	"github.com/HyperloopUPV-H8/Backend-H8/vehicle/models"
 	"github.com/rs/zerolog"
 	trace "github.com/rs/zerolog/log"
 )
 
+const DEFAULT_ORDER = 100
+
 type UpdateFactory struct {
-	count     map[uint16]uint64
-	timestamp map[uint16]uint64
-	trace     zerolog.Logger
+	count        map[uint16]uint64
+	cycleTimeAvg map[uint16]*common.MovingAverage[uint64]
+	timestamp    map[uint16]uint64
+	fieldAvg     map[uint16]map[string]*common.MovingAverage[float64]
+	trace        zerolog.Logger
 }
 
 func NewFactory() UpdateFactory {
 	trace.Info().Msg("new update factory")
 	return UpdateFactory{
-		count:     make(map[uint16]uint64),
-		timestamp: make(map[uint16]uint64),
-		trace:     trace.With().Str("component", "updateFactory").Logger(),
+		count:        make(map[uint16]uint64),
+		cycleTimeAvg: make(map[uint16]*common.MovingAverage[uint64]),
```

Comment:
Why are the moving averages pointers instead of values?

Reply:
to prevent copies, because methods of updateFactory use it as a value instead of a reference, which can create copies of the value